app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java

```diff
@@ -108,6 +113,39 @@ public void onCreatePreferences(Bundle bundle, String rootKey) {
             return true;
         });
 
+        ListPreference countryPreference = ((ListPreference) findPreference("user_country"));
+        List<String> countryLabels = new ArrayList<>();
+
+        DaoSession daoSession = OFFApplication.getInstance().getDaoSession();
+        AsyncSession asyncSessionCountries = daoSession.startAsyncSession();
+        CountryNameDao countryNameDao = daoSession.getCountryNameDao();
+
+        asyncSessionCountries.queryList(countryNameDao.queryBuilder()
+                .where(CountryNameDao.Properties.LanguageCode.eq(LocaleHelper.getLanguage(getActivity())))
+                .orderAsc(CountryNameDao.Properties.Name).build());
+
+        asyncSessionCountries.setListenerMainThread(operation -> {
+            @SuppressWarnings("unchecked")
+            List<CountryName> countryNames = (List<CountryName>) operation.getResult();
+            for (int i = 0; i < countryNames.size(); i++) {
+                countryLabels.add(countryNames.get(i).getName());
+            }
+            countryPreference.setEntries(countryLabels.toArray(new String[countryLabels.size()]));
+            countryPreference.setEntryValues(countryLabels.toArray(new String[countryLabels.size()]));
+        });
+
+        countryPreference.setOnPreferenceChangeListener(((preference, newValue) -> {
+            if (preference instanceof ListPreference) {
+               if (preference.getKey().equals("user_country")) {
```

Comment:
Could you extract out "user_country" key into a constant so that it can be referenced.

Reply:
Right, I'll do it right away.

Reply:
The key has been extracted as a static constant which can be referenced.